src/server/string_family.cc

```diff
@@ -646,6 +646,42 @@ Responder* StringFamily::TestResponder2(CmdArgList args, ConnectionContext* cntx
   return rsp;
 }
 
+void StringFamily::TestResponder3(CmdArgList args, ConnectionContext* cntx) {
+  Transaction* tx = cntx->transaction;
+  tx->StartMultiGlobal(tx->GetDbIndex());
+
+  // This example shows we can store responders and send their replies later.
+  // It also shows, how we can potentially stall for big responses.
+  vector<char[64]> bufs(10);
+  vector<Responder*> rsps;
+  for (int i = 0; i < 10; i++) {
+    cntx->SetResponderBuffer(bufs[i]);
+
+    tx->MultiSwitchCmd(tx->GetCId());
+    tx->InitByArgs(tx->GetDbIndex(), args);
+
+    auto* rsp = cntx->MakeResponder<AtomicCounterResponder>(tx);
+    tx->ScheduleSingleHop([rsp, i](Transaction* t, EngineShard* sd) {
```

Comment:
this one is blocking, is not it?

Reply:
does it mean we need `ScheduleSingleHop` asynchronous version ?

Reply:
I'm not sure what to use, currently cid->opt_mask marks itself as async if the handler has the right return type